packages/orchestrator-ui-components/src/components/WfoTable/WfoAdvancedTable/WfoAdvancedTable.tsx

```diff
@@ -120,18 +98,17 @@ export const WfoTableWithFilter = <T extends object>({
         },
     };
 
-    const tableColumnsWithControlColumns: WfoBasicTableColumnsWithControlColumns<T> =
-        {
-            ...leadingControlColumns,
-            ...tableColumns,
-            ...trailingControlColumns,
-            ...(detailModal ? detailsIconColumn : []),
-        };
+    const tableColumnsWithControlColumns: WfoAdvancedTableColumnConfig<T> = {
+        ...(detailModal && detailsIconColumn),
+        ...tableColumnConfig,
+    };
+
+    const tableSettingsColumns: TableSettingsColumnConfig<T>[] = Object.entries(
+        tableColumnConfig,
+    ).map((keyValuePair) => {
```

Comment:
can't you destructure here? Something like:

```suggestion
    ).map(([key, value]) => {
```

Reply:
done